pkg/drupal/dbadmincheck.go

```diff
@@ -0,0 +1,139 @@
+package drupal
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os/exec"
+	"strings"
+
+	"github.com/salsadigitalauorg/shipshape/pkg/shipshape"
+	"github.com/salsadigitalauorg/shipshape/pkg/utils"
+)
+
+const AdminUser shipshape.CheckType = "drupal-admin-user"
+
+// AdminUserCheck fetches all role configurations from the database and verifies
+// they do not have is_admin set to true.
+type AdminUserCheck struct {
+	shipshape.CheckBase `yaml:",inline"`
+	DrushCommand        `yaml:",inline"`
+	// List of role names allowed to have is_admin set to true.
+	AllowedRoles []string `yaml:"allowed-roles"`
+	userRoles    []string
+	roleConfigs  map[string]bool
+}
+
+type roleConf struct {
+	IsAdmin bool `json:"is_admin"`
+	Name string `json:"id"`
+}
+
+// Init implementation for the drush-based user role config check.
+func (c *AdminUserCheck) Init(ct shipshape.CheckType) {
+	c.CheckBase.Init(ct)
+	c.RequiresDb = true
+}
+
+// Merge implementation for AdminUserCheck check.
+func (c *AdminUserCheck) Merge(mergeCheck shipshape.Check) error {
+	adminUserMergeCheck := mergeCheck.(*AdminUserCheck)
+	if err := c.CheckBase.Merge(&adminUserMergeCheck.CheckBase); err != nil {
+		return err
+	}
+
+	c.DrushCommand.Merge(adminUserMergeCheck.DrushCommand)
+	utils.MergeStringSlice(&c.AllowedRoles, adminUserMergeCheck.AllowedRoles)
+	return nil
+}
+
+// getActiveRoles runs the drush command to populate data for the roles config check.
+func (c *AdminUserCheck) getActiveRoles() map[string]string {
+  var err error
+
+	activeRoles := map[string][]byte{}
```

Comment:
I think we don't need the map here - `[]byte{}` should work.
```suggestion
	activeRoles := []byte{}
```